CMakeLists.txt

```diff
@@ -511,7 +512,7 @@ if(USE_PYTHON)
 		DEPENDS qulacs_core
 		COMMAND pytest -v -s ${CMAKE_CURRENT_SOURCE_DIR}/python/tests
 	)
-endif()
 
 #dependency setting for ExternalProject
 add_dependencies(qulacs_core eigen)
+endif()
```

Comment:
I think that the targets `qulacs_core` and `eigen` are not visible in `python/CMakeLists.txt` so the dependency needs to be handled in the main `./CMakeLists.txt` file. 

It might be more readable to ident the two lines which I did in my forked repository. Please let me know if I should change anything else in the PR.

Reply:
> I think that the targets qulacs_core and eigen are not visible in python/CMakeLists.txt

The `qulacs_core` library is defined in this line of `python/CMakeLists.txt`.
https://github.com/qulacs/qulacs/blob/d6306cecc754f315e87cd4b2ee53d592528ae28c/python/CMakeLists.txt#L36

The `eigen` is defined in `./CMakeLists.txt`, but when `USE_PYTHON=Yes`, the subdirectory is loaded, so it is possible to find the dependency.

https://github.com/qulacs/qulacs/blob/d6306cecc754f315e87cd4b2ee53d592528ae28c/CMakeLists.txt#L415-L417

For example, `pybind11_pop` is added as a dependency in `python/CMakeLists.txt`.

https://github.com/qulacs/qulacs/blob/d6306cecc754f315e87cd4b2ee53d592528ae28c/python/CMakeLists.txt#L42

What do you think?

Reply:
I fully agree. How would you like me to change the PR? Shall I only move the `add_dependencies(qulacs_core eigen)` to `python/CMakeLists.txt` or everything that is needed for PyBind11 (which includes fetching the eigen library)?

Reply:
Please can you move only `add_dependencies(qulacs_core eigen)` to `python/CMakeLists.txt`? I think that should work fine. Thank you!

Reply:
Done so. I you want me to make any additional changes before merging this PR please let me know.

Reply:
I also checked that moving `add_dependencies(qulacs_core eigen)` to `python/CMakeLists.txt` works well.